### PR TITLE
WIP Basic matchAttempt vs a simple RH swar block.  

### PR DESCRIPTION
**Pull request by Scottbruceheart**
This is an ok impl.
I'd prefer to use the sublanes abstraction a bit more cleanly, and I feel like we're missing a trick with equality testing.
Added greaterEqualZeroLeft (since we call that in a few places, though opt compiler probably catches that, note we still don't have a general greaterEqual)
Added some basic tests for matchAttempt.

### CODE REVIEW

File: inc/zoo/map/RobinHood_util.h [hunk 1]
```diff
@@ -1,5 +1,7 @@
 #pragma once
 
+#include "zoo/meta/BitmaskMaker.h"
+#include "zoo/meta/log.h"
```
Comment (thecppzoo) [inline inc/zoo/map/RobinHood_util.h:4]: These come from `SWAR.h`

File: inc/zoo/swar/SWAR.h [hunk 2]
```diff
@@ -256,16 +256,29 @@ struct BooleanSWAR: SWAR<NBits, T> {
...
+    template<int NB, typename TT> friend
+    constexpr BooleanSWAR<NB, TT> differents(SWAR<NB, TT> left, SWAR<NB, TT> right) noexcept;
 
 };
 
+// Differents becuase "differences of lanes in boolean bits"
+template<int NB, typename TT> constexpr
+BooleanSWAR<NB, TT> differents(SWAR<NB, TT> left, SWAR<NB, TT> right) noexcept {
+    return left ^ right;
+}
```
Comment (thecppzoo) [inline inc/zoo/swar/SWAR.h:275]: This does not leave the most significant bit set on "true", the MSB must be the boolean, and the rest 0

File: inc/zoo/map/RobinHood_util.h (hunk 1 above)
Comment (Scottbruceheart) [reply inc/zoo/map/RobinHood_util.h]: fixed.

**Comment by Scottbruceheart** [comment]
This works for basic match, but your provider abstraction is better.

**Comment by Scottbruceheart** [comment]
This is too old, closing.

**Pull request closed without merging**